### PR TITLE
Feature/activity UI

### PR DESCRIPTION
**Pull request by Feez2403**
- Created The List of activites on the List Fragement
- Created The activity Activity
- Clicking on an activity in the list opens the corresponding activity 
- Added corresponding tests to everything above
- Added more activites under ActivitiesQuerryHelper to show the Recycler in action
- Added an helper function to Refresh a fragment
- (Requested by mr Rygone and I totally approve) Tiny refactor of LoginFragment to the user setter

Thanks in advance for your reviews and feel free to tell me if you find any improvement we can do !

### CODE REVIEW

File: app/src/main/res/layout/activity_activity.xml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        android:text="@string/text_lorem_ipsum"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/textView7" />
+
+    <TextView
+        android:id="@+id/txt_activity_organizer"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="organizer"
```
Comment (AlessioVerardo) [inline app/src/main/res/layout/activity_activity.xml:94]: We should probably put this in the string resources

File: app/src/main/res/layout/activity_activity.xml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintHorizontal_bias="0.502"
+        app:layout_constraintStart_toEndOf="@+id/img_activity_logo"
+        app:layout_constraintTop_toTopOf="@+id/img_activity_logo" />
+
+    <TextView
+        android:id="@+id/txt_activity_zone"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="zone"
```
Comment (AlessioVerardo) [inline app/src/main/res/layout/activity_activity.xml:47]: We should probably put this in the string resources

File: app/src/main/res/layout/activity_activity.xml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/img_activity_logo"
+        app:layout_constraintTop_toTopOf="@+id/img_activity_logo"
+        app:layout_constraintVertical_bias="0.552" />
+
+    <TextView
+        android:id="@+id/txt_activity_date"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="date"
```
Comment (AlessioVerardo) [inline app/src/main/res/layout/activity_activity.xml:36]: We should probably put this in the string resources

File: app/src/main/res/layout/activity_activity.xml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent"
+        app:srcCompat="@android:mipmap/sym_def_app_icon"
+        tools:srcCompat="@tools:sample/avatars" />
+
+    <TextView
+        android:id="@+id/txt_activity_Name"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="Title"
```
Comment (AlessioVerardo) [inline app/src/main/res/layout/activity_activity.xml:24]: We should probably put this in the string resources

File: app/src/main/java/com/github/sdpteam15/polyevents/fragments/ListFragment.kt [hunk 5]
```diff
@@ -1,34 +1,76 @@
...
+import com.github.sdpteam15.polyevents.helper.ActivitiesQueryHelper
+import com.github.sdpteam15.polyevents.helper.ActivitiesQueryHelperInterface
+import com.github.sdpteam15.polyevents.helper.HelperFunctions
 
-// TODO: Rename parameter arguments, choose names that match
-// the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
-private const val ARG_PARAM1 = "param1"
-private const val ARG_PARAM2 = "param2"
+
+const val EXTRA_ACTIVITY = "com.github.sdpteam15.polyevents.activity.ACTIVITY_ID"
```
Comment (Rygone) [inline app/src/main/java/com/github/sdpteam15/polyevents/fragments/ListFragment.kt:21]: A comment is missing to explain this constant.

File: app/src/androidTest/java/com/github/sdpteam15/polyevents/ActivityActivityTest.kt [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+class ActivityActivityTest {
+    lateinit var activities: ArrayList<Activity>
+    lateinit var mockedUpcomingActivitiesProvider: ActivitiesQueryHelperInterface
+
+    @Rule
+    @JvmField
+    var mainActivity = ActivityScenarioRule(MainActivity::class.java)
+
+    @Before
+    fun setup() {
```
Comment (Rygone) [inline app/src/androidTest/java/com/github/sdpteam15/polyevents/ActivityActivityTest.kt:48]: good idea the setup function

**Review by Rygone — APPROVED** [review]

File: app/src/main/java/com/github/sdpteam15/polyevents/fragments/ListFragment.kt (hunk 5 above)
Comment (Feez2403) [reply app/src/main/java/com/github/sdpteam15/polyevents/fragments/ListFragment.kt:21]: Thank you, I'll change it in the next commit

File: app/src/main/res/layout/activity_activity.xml [hunk 7]
```diff
@@ -0,0 +1,99 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".ActivityActivity">
```
Comment (Mathematou) [inline app/src/main/res/layout/activity_activity.xml:7]: It would be nice in the future to change to a scrollable layout

**Review by Mathematou — APPROVED** [review]

**Pull request merged**